src/privatesend-client.cpp

```diff
@@ -1177,6 +1177,10 @@ bool CPrivateSendClient::MakeCollateralAmounts(const CompactTallyItem& tallyItem
 {
     LOCK2(cs_main, pwalletMain->cs_wallet);
 
+    // denominated input is always a single one, so we can check its amount directly and return early
+    if(!fTryDenominated && tallyItem.vecTxIn.size() == 1 && pwalletMain->IsDenominatedAmount(tallyItem.nAmount))
```

Comment:
It seems that `IsDenominatedAmount` doesn't require synchronization, so probably it would be safe to move this check above `LOCK2`